screeps-game-api/src/objects/impls/room.rs

```diff
@@ -420,3 +435,228 @@ pub enum Path {
+                let mut obj_id = None;
+                let mut data = None;
+                let mut data_buffer: Option<serde_json::Value> = None;
+
+                while let Some(key) = map.next_key()? {
+                    match key {
+                        Field::Event => {
+                            if event_type.is_some() {
+                                return Err(de::Error::duplicate_field("event"));
+                            }
+                            event_type = Some(map.next_value()?);
+                        }
+                        Field::ObjectId => {
+                            if obj_id.is_some() {
+                                return Err(de::Error::duplicate_field("objectId"));
+                            }
+                            obj_id = Some(map.next_value()?);
+                        }
+                        Field::Data => {
+                            if data.is_some() {
+                                return Err(de::Error::duplicate_field("data"));
+                            }
+
+                            match event_type {
+                                None => data_buffer = map.next_value()?,
+                                Some(event_id) => {
+                                    data = match event_id {
+                                        1 => Some(EventType::Attack(map.next_value()?)),
+                                        2 => Some(EventType::ObjectDestroyed(map.next_value()?)),
+                                        3 => Some(EventType::AttackController),
+                                        4 => Some(EventType::Build(map.next_value()?)),
+                                        5 => Some(EventType::Harvest(map.next_value()?)),
+                                        6 => Some(EventType::Heal(map.next_value()?)),
+                                        7 => Some(EventType::Repair(map.next_value()?)),
+                                        8 => Some(EventType::ReserveController(map.next_value()?)),
+                                        9 => Some(EventType::UpgradeController(map.next_value()?)),
+                                        10 => Some(EventType::Exit(map.next_value()?)),
+                                        _ => {
+                                            return Err(de::Error::custom(format!(
+                                                "Event Type Unrecognized: {}",
+                                                event_id
+                                            )))
+                                        }
+                                    };
+                                }
+                            };
+                        }
+                    }
+                }
+
+                if data_buffer.is_some() && event_type.is_some() && data.is_none() {
+                    let val: serde_json::Value = data_buffer.unwrap();
+                    let err = |_| de::Error::custom("Can't parse Event data.");
```

Comment:
Could we include the display representation of the original error here? It'd be much more useful than a static string. Just something like `de::Error::custom(format_args!("can't parse event data due to inner error {}", e))` could work? We should also avoid capitalization and an ending period as serde inserts this as part of a larger sentence (or at least that's what [custom](https://docs.rs/serde/1.0.79/serde/de/trait.Error.html#tymethod.custom)'s documentation states?).